packages/core/src/processors/runner.ts

```diff
@@ -0,0 +1,291 @@
+    }
+    this.allChunks.push(chunk);
+  }
+
+  // Public interface for processors
+  getState(): Record<string, any> {
+    return {
+      ...Object.fromEntries(this.customState),
+    };
+  }
+
+  // Helper methods for backward compatibility
+  get<T>(key: string): T | undefined {
+    return this.customState.get(key);
+  }
+
+  set<T>(key: string, value: T): void {
+    this.customState.set(key, value);
+  }
+}
+
+export class ProcessorRunner {
+  public readonly inputProcessors: Processor[];
+  public readonly outputProcessors: Processor[];
+  private readonly logger: IMastraLogger;
+  private readonly agentName: string;
+
+  constructor({
+    inputProcessors,
+    outputProcessors,
+    logger,
+    agentName,
+  }: {
+    inputProcessors?: Processor[];
+    outputProcessors?: Processor[];
+    logger: IMastraLogger;
+    agentName: string;
+  }) {
+    this.inputProcessors = inputProcessors ?? [];
+    this.outputProcessors = outputProcessors ?? [];
+    this.logger = logger;
+    this.agentName = agentName;
+  }
+
+  async runOutputProcessors(messageList: MessageList, telemetry?: any): Promise<MessageList> {
+    const responseMessages = messageList.clear.response.v2();
+
+    let processableMessages: MastraMessageV2[] = [...responseMessages];
+
+    const ctx: { messages: MastraMessageV2[]; abort: () => never } = {
+      messages: processableMessages,
+      abort: () => {
+        throw new TripWire('Tripwire triggered');
+      },
+    };
+
+    for (const [index, processor] of this.outputProcessors.entries()) {
+      const abort = (reason?: string): never => {
+        throw new TripWire(reason || `Tripwire triggered by ${processor.name}`);
+      };
+
+      ctx.abort = abort;
+
+      // Use the processOutputResult method if available
+      const processMethod = processor.processOutputResult?.bind(processor);
+
+      if (!processMethod) {
+        // Skip processors that don't implement processOutputResult
+        continue;
+      }
+
+      if (!telemetry) {
+        processableMessages = await processMethod({ messages: processableMessages, abort: ctx.abort });
+      } else {
+        await telemetry.traceMethod(
+          async () => {
+            processableMessages = await processMethod({ messages: processableMessages, abort: ctx.abort });
+            return processableMessages;
+          },
+          {
+            spanName: `agent.outputProcessor.${processor.name}`,
+            attributes: {
+              'processor.name': processor.name,
+              'processor.index': index.toString(),
+              'processor.total': this.outputProcessors.length.toString(),
+            },
+          },
+        )();
+      }
+    }
+
+    if (processableMessages.length > 0) {
+      messageList.add(processableMessages, 'response');
+    }
+
+    return messageList;
+  }
+
+  /**
+   * Process a text chunk through all output processors with state management
+   */
+  async processChunk(
```

Comment:
should probably rename chunk to part

Reply:
yeah was thinking this too, just hadn't gotten around to that, i'll add to TODOs